### ISSUE TITLE
Multiple QuAMs causing issues with references

### ISSUE THREAD

**Issue by JacobHast** [qua-platform/quam#58]
When initializing multiple quam roots in a single script something appears to go wrong with referencing. I'm not sure when it makes sense to initialize multiple quams, but in my case this has caused some hard-to-find bugs when testing various things out. If initializing multiple roots is not allowed, maybe a suggestion would be to raise an error if it happens.

![image](https://github.com/qua-platform/quam/assets/77778159/dbd9f382-b506-4f6d-a641-12bb29127a0a)

**Comment by nulinspiratie** [comment]
You're right, creating multiple QuamRoot objects causes issues because `QuamBase._root` is always set to the latest QuAM, which is needed for absolute referencing to work.
However, I don't see a straightforward solution for this. There are several situations where one would create a new QuamRoot object. For instance, if one is running a live kernel and loads QuAM multiple times.

I will keep this open, but for now I don't see a solution that wouldn't introduce significant overhead. @JacobHast if you have suggestions please let me know

**Comment by JacobHast** [comment]
I see. Would it be possible to add a warning when creating a new root if a root already exists to facilitate debugging?

**Comment by nulinspiratie** [comment]
At least for now I wouldn't want it as a warning because that usually shows up by default. I could do `logging.info`, which will or will not show up depending on the log filtering. Does that work?

**Comment by JacobHast** [comment]
Hi Serwan, this one is still causing issues for us, especially for new users playing around with creating new QuAM instances without restarting the kernel. As you mentioned, when loading a QuAM multiple times there doesn't seem to be an issue, and so you dont want to warn the user in that case. Is it possible to create a warning that doesn't trigger on loads, but does trigger on creating multiple new QuAMs from scratch?

**Comment by nulinspiratie** [comment]
It could probably be done, but let me make sure I understand the underlying issue properly before implementing:

You're saying that when a QuAM is loaded multiple times there isn't an issue, can you clarify why? I would imagine that if you're using a qubit from the previous quam, its parents ancestry would still point to the old QuAM, but any references would point to the new QuAM. This seems like it's still an issue.

One alternative solution that I had in mind was that when a new QuAM is initialized, it will check if `QuamBase._root` is defined, if so, it will get all its components (`QuamBase._root.get_components`), and for each component it set the instance variables (not the class variable) `_root` and the `parent` to None. This should at least partially disable all "old" components.
Would this solve the issue?

**Comment by JacobHast** [comment]
Actually, we just got an issue because a user loaded a QuAM state multiple times in the same kernel. Ideally I would love a solution where multiple QuAMs can be loaded. Maybe QuamBase._root needs to be a list? Your suggestion with disabling old QuAM might also solve some issues, but I imagine this can still cause trouble, e.g. if the user want to run multiple experiments on both QuAMs in the same kernel. In any case, my top request would be that some kind of warning is displayed when loading multiple QuAMs to at least facilitate debugging until a stable solution is found.

**Comment by nulinspiratie** [comment]
Yes agreed, raising a warning is the most critical step now. I've added a warning in #103, please let me know if it's filtered out.

I also think I've found a way to solve this problem by getting rid of `_root` altogether. It seems to work (see branch `feat/remove-root` if you want to try it out). However, the remaining step is implementing this without any breaking changes. There are some edge cases that need to be handled

**Comment by JacobHast** [comment]
The warning looks good here.

And exciting to hear that a permanent solution might be close!
